Assets/MRTK/Core/Providers/InputSimulation/SimulatedMotionController.cs

```diff
@@ -0,0 +1,173 @@
+        /// <remarks>A single interaction mapping works for both left and right controllers.</remarks>
+        public override MixedRealityInteractionMapping[] DefaultInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer),
+            new MixedRealityInteractionMapping(1, "Spatial Grip", AxisType.SixDof, DeviceInputType.SpatialGrip),
+            new MixedRealityInteractionMapping(2, "Grip Press", AxisType.SingleAxis, DeviceInputType.TriggerPress),
+            new MixedRealityInteractionMapping(3, "Trigger Position", AxisType.SingleAxis, DeviceInputType.Trigger),
+            new MixedRealityInteractionMapping(4, "Trigger Touch", AxisType.Digital, DeviceInputType.TriggerTouch),
+            new MixedRealityInteractionMapping(5, "Trigger Press (Select)", AxisType.Digital, DeviceInputType.Select),
+            new MixedRealityInteractionMapping(6, "Touchpad Position", AxisType.DualAxis, DeviceInputType.Touchpad),
+            new MixedRealityInteractionMapping(7, "Touchpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch),
+            new MixedRealityInteractionMapping(8, "Touchpad Press", AxisType.Digital, DeviceInputType.TouchpadPress),
+            new MixedRealityInteractionMapping(9, "Menu Press", AxisType.Digital, DeviceInputType.Menu),
+            new MixedRealityInteractionMapping(10, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick),
+            new MixedRealityInteractionMapping(11, "Thumbstick Press", AxisType.Digital, DeviceInputType.ThumbStickPress),
+        };
+
+        internal void UpdateState(SimulatedMotionControllerData motionControllerData)
+        {
+            lastPose = currentPose;
+            currentPose.Position = motionControllerData.Position;
+            currentPose.Rotation = motionControllerData.Rotation;
+            IsPositionAvailable = IsRotationAvailable = motionControllerData.Position != Vector3.zero;
+
+            if (lastPose != currentPose)
+            {
+                if (IsPositionAvailable && IsRotationAvailable)
+                {
+                    CoreServices.InputSystem?.RaiseSourcePoseChanged(InputSource, this, currentPose);
+                }
+            }
+
+            for (int i = 0; i < Interactions?.Length; i++)
+            {
+                switch (Interactions[i].InputType)
+                {
+                    case DeviceInputType.SpatialPointer:
+                        Interactions[i].PoseData = currentPose;
+                        if (Interactions[i].Changed)
+                        {
+                            CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction, currentPose);
+                        }
+                        break;
+                    case DeviceInputType.SpatialGrip:
+                        Interactions[i].PoseData = currentPose;
```

Comment:
Checking my reading - SpatialPointer and SpatialGrip are handled the same way (i..e these 5 lines of code are the same right?)

Reply:
That's right - redundant code removed.